mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAOptimizer.scala

```diff
@@ -119,6 +120,14 @@ final class EMLDAOptimizer extends LDAOptimizer {
       }
     }
 
+    // create term vertices for empty docs
+    val emptyDocTermVertices: RDD[(VertexId, TopicCounts)] =
+      docs.filter { case (_, termCounts: Vector) =>
+        Vectors.norm(termCounts, p = 1.0) == 0.0 // empty docs has all zero termCounts
+      }.mapValues { case(termCounts: Vector) =>
```

Comment:
I like specifying the type for emptyDocTermVertices since it's pretty complex, but I agree about the other simplifications.  Style call